### DISCUSSION TITLE
@QueryMapping annotation not working

### DISCUSSION THREAD

**Discussion by reenaupadhyay** [graphql-java/graphql-java#3216] [Q&A] [answered]
Hi folks,

I am new to graphQl. I have just purchased "GraphQL with Java and Spring" book by Andi and Donna. Followed the exact same steps mentioned in the Introduction chapter for building first graphQl service. Here's the code- 
https://github.com/reenaupadhyay/graphql-service-demo

spring boot version used is  - 3.0.4 with java 17
I started the spring boot application, but the query is returning null in response.
URL:
http://localhost:8080/graphiql?path=/graphql
Query fired on above URL:
query myPets {
pets {
name
color
}
}
 
Response returned is:
{
  "data": {
    "pets": null
  }
}

I have hardcoded the Pets response in the PetsController, but looks like @QueryMapping annotation is not working.

![image](https://github.com/graphql-java/graphql-java/assets/12643031/a5c778ac-021f-4689-94c8-625b4d29bdb9)

**Comment by bclozel** [comment, marked as answer]
Hello @reenaupadhyay 

I really enjoyed reading the book, hope you like it as well.
I think the controller class is not located in the right package. You should organize all packages under the main class as suggested here https://docs.spring.io/spring-boot/docs/current/reference/html/using.html#using.structuring-your-code.locating-the-main-class

In your case, the controller class is not scanned, so the controller methods are not detected.

**Reply by reenaupadhyay** [reply]
Thank you. It works after organizing the classes in the correct package.
